src/cdk-experimental/testing/protractor/protractor-element.ts

```diff
@@ -50,4 +50,9 @@ export class ProtractorElement implements TestElement {
   async getAttribute(name: string): Promise<string|null> {
     return this.element.getAttribute(name);
   }
+
+  async hasClass(name: string): Promise<boolean> {
+    const classes = (await this.getAttribute('class')) || '';
+    return new Set(classes.split(/\s+/).filter(c => c)).has(name);
```

Comment:
Does this need to `trim` on each item as well?

Reply:
Splitting on `/\s+/` should act as an auto-trim, the only concern would be empty strings getting into the set, but the `filter(c => c)` takes care of that